### PR TITLE
[fix](partial update) correct col_nums when init agg state in memtable

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

File: regression-test/data/unique_with_mow_p0/partial_update/basic_with_duplicate2.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+1,666
+1,777
```
Comment (zhannngchen) [inline regression-test/data/unique_with_mow_p0/partial_update/basic_with_duplicate2.csv:2]: we'd better use data that out of order, which can cover more cases

**Comment by bobhan1** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by bobhan1** [comment]
run clickbench

**Pull request merged**